datafusion/physical-expr/src/expressions/binary/kernels_arrow.rs

```diff
@@ -333,34 +337,88 @@ pub(crate) fn add_dyn_temporal(left: &ArrayRef, right: &ArrayRef) -> Result<Arra
     }
 }
 
-pub(crate) fn add_dyn_temporal_scalar(
+pub(crate) fn add_dyn_temporal_right_scalar(
```

Comment:
Since adding `interval + duration` is the same as adding `duration + interval` I wonder if we need both of these kernels? Instead, perhaps we can use the same pattern as used elsewhere in binary.rs and do something like:

Given: `a + b`:
* if `b` is a scalar, call `add_dyn_scalar(a, b)`
* if `a` is a scalar, call `add_dyn_scalar(b, a)`

Given: `a - b`:
* if `b` is a scalar, call `sub_dyn_scalar(a, b)`
* if `a` is a scalar, call `add_dyn_scalar(-b, a)`

🤔

Reply:
A question that comes to my mind about this approach is whether the unary minus on the LHS (the array) would result in unnecessary computation or not. Given the complexity of the types involved, I am not sure if the compiler would be smart enough to optimize things all the way and emit simple subtraction code (vs. negating the entire array and then summing again).